openweather/src/main/java/pages/SearchPage.java

```diff
@@ -25,16 +28,40 @@ public SearchPage(WebDriver driver) {
     }
 
     // TODO: use a common method to get row/cell text
-    public List<CityInfo> getSearchResult() {
+    public List<CityInfo>  getSearchResult() {
         List<CityInfo> cities = new ArrayList<>();
-        for (WebElement row : getTableRows()) {
-            String name = row.findElement(By.tagName("a")).getText();
-            String temperature =
-                    row.findElement(By.xpath("//*[@class='badge badge-info'")).getText();
-            cities.add(new CityInfo(name, temperature));
-        }
+            for (WebElement row : getTableRows()) {
+                String name = row.findElement(By.tagName("a")).getText();
+                String temperature =
+                        row.findElement(By.xpath("//span[@class='badge badge-info']")).getText();
+                cities.add(new CityInfo(name, temperature));
+            }
 
         return cities;
+   }
+
+    public void checkSearchResult(String searchCity) {
+        List<CityInfo> cities = getSearchResult();
+        boolean searchResult=true;
+        String msg="List of search city is returned";
+        SoftAssert softAssertion= new SoftAssert();
+
+        if (searchCity != "Not Found" && cities.size() > 0) {
+            for (CityInfo city : cities) {
+                if (!city.getCityName().contains(searchCity)) {
+                    searchResult = false;
+                    msg="Search results do not contain city";
+                }
+            }
+        } else if (searchCity != "Not Found" && cities.size() == 0) {
+            searchResult = false;
+            msg="Search result is Not Found while expectation is list of cities";
+        } else if (searchCity == "Not Found" && cities.size() > 0) {
+            searchResult = false;
+            msg="Search result is the list of cities while expectation is Not Found";
+        }
+        softAssertion.assertEquals(searchResult,true,msg);
```

Comment:
Be very careful with soft assert. I will always need to fail fast when having a failure. If sth you are expecting is not satisfied, running remaining code is wasting time and resources.